packages/features/services/firebase.ts

```diff
@@ -0,0 +1,62 @@
+import { initializeApp } from 'firebase/app';
+import { getAnalytics, Analytics } from 'firebase/analytics';
+import { getFirestore, collection, getDocs, addDoc } from 'firebase/firestore';
+
+import firebaseConfig from '@packages/config/firebase';
+
+import type { Member } from '@packages/entities/members';
+import type { Result } from '@packages/utils/functions';
+
+const app = initializeApp(firebaseConfig);
+const db = getFirestore(app);
+
+export function getFirebaseWebServices(): FirebaseWebServices {
+  // Initialize Firebase services
+  const analytics = getAnalytics(app);
+  return { analytics };
+}
+
+export function getFirebaseApiServices(): FirebaseApiServices {
+  const memberCollection = collection(db, 'members');
+
+  async function addMembers(member: Member): Promise<Result<AddMemberResp>> {
+    try {
+      const docRef = await addDoc(memberCollection, member);
+      const resp: AddMemberResp = { id: docRef.id };
+
+      return [resp, null];
+    } catch (e) {
+      return [null, e as Error];
+    }
+  }
+
+  async function getMembers(): Promise<Result<Member[]>> {
+    try {
+      const querySnapshot = await getDocs(memberCollection);
+      const resp = querySnapshot.docs.map((doc) => doc.data() as Member);
+
+      return [resp, null];
+    } catch (error) {
+      return [null, error as Error];
+    }
+  }
+
+  return {
+    db: { getMembers, addMembers },
+  };
+}
+
+export interface FirebaseWebServices {
+  analytics: Analytics;
+}
+
+interface FirebaseApiServices {
```

Comment:
não seria legal mover essas interfaces para o inicio do arquivo ? particularmente acho que a leitura fica mais fluida

Reply:
Bom, a ideia é que tudo que vai ser exportado venha antes. Essa interface só é usada aqui dentro, então acho que faz sentido ela ficar mais pro fim do arquivo.

Quando abrimos um arquivo, queremos ler o primeiramente o que ele exporta, ou seja, o que são metodos publicos. Se vc for pensar, a interface tem prioridade secundária se comparada ao código